Building with make 4.3

Building Embox with make 4.3 causes following error:

```
> make
 MKGEN mk_core_def.mk
mk/core/define.mk:1463: *** recipe commences before first target.  Stop.
make[2]: *** [mk/load-mk.inc.mk:165: mk/.cache/mk/mk_core_def.mk] Error 2
make[2]: *** Deleting file 'mk/.cache/mk/mk_core_def.mk'
make[1]: *** [mk/main.mk:30: build] Error 2
make: *** [Makefile:37: all] Error 2
```

Temporary solution is to downgrade `make` to 4.2.

Seems like the problem is caused by this function in `mk/core/define.mk` at line 1379: `__builtin_to_function_inline`.

--- a/Makefile
+++ b/Makefile
@@ -15,11 +15,6 @@ $(error Unsupported Make version. \
 	please use GNU Make 3.81 or above.)
 endif
 
-ifeq (4.3, $(MAKE_VERSION))
-$(error Make 4.3 is not supported. \
-	Please consider downgrading to 4.2 or using newer version)
-endif
-
 # Disable everything, turn on undefined variables check.
 MAKEFLAGS += --no-builtin-rules
 MAKEFLAGS += --no-builtin-variables
@@ -39,5 +34,5 @@ export PS1 :=
 
 .PHONY : $(sort all $(MAKECMDGOALS))
 $(sort all $(MAKECMDGOALS)) :
-	@$(MAKE) -C $(dir $(lastword $(MAKEFILE_LIST))) -f mk/main.mk $@
+	@ulimit -s 65536 && $(MAKE) -C $(dir $(lastword $(MAKEFILE_LIST))) -f mk/main.mk $@
 

--- a/mk/core/define.mk
+++ b/mk/core/define.mk
@@ -1264,6 +1264,9 @@ define builtin_func-def-ifndef
 	$(if $(value $(call __def_expand,$1)),$(value 3),$2)
 endef
 
+# Builtin def-ifdef/def-ifndef macros are ready to be used.
+$(def_all)
+
 #
 # Builtin to user-defined function call converters.
 #

--- a/mk/extbld/lib.mk
+++ b/mk/extbld/lib.mk
@@ -79,14 +79,14 @@ download : $(DOWNLOAD) $(DOWNLOAD_CHECK)
 EXTRACT  := $(BUILD_DIR)/.extracted
 extract : $(EXTRACT)
 $(EXTRACT): $(DOWNLOAD) | $(DOWNLOAD_DIR) $(BUILD_DIR)
-	$(if $(filter %zip,$(pkg_ext)), \
-		unzip -q $(DOWNLOAD_DIR)/$(pkg_archive_name) -d $(BUILD_DIR), \
-		tar -xf $(DOWNLOAD_DIR)/$(pkg_archive_name) -C $(BUILD_DIR))
+	$(if $(first_url),$(if $(filter %zip,$(pkg_ext)), \
+		unzip -q $(DOWNLOAD_DIR)/$(pkg_archive_name) -d $(BUILD_DIR);, \
+		tar -xf $(DOWNLOAD_DIR)/$(pkg_archive_name) -C $(BUILD_DIR);) \
 	COPY_FILES="$(addprefix $(DOWNLOAD_DIR)/, \
 			$(call targets_git,$(sources_git)))"; \
 		if [ "$$COPY_FILES" ]; then \
 			cp -R $$COPY_FILES $(BUILD_DIR); \
-		fi
+		fi; )
 	touch $@
 
 PATCH  := $(BUILD_DIR)/.patched

--- a/mk/extbld/lib.mk
+++ b/mk/extbld/lib.mk
@@ -78,7 +78,7 @@ download : $(DOWNLOAD) $(DOWNLOAD_CHECK)
 
 EXTRACT  := $(BUILD_DIR)/.extracted
 extract : $(EXTRACT)
-$(EXTRACT): | $(DOWNLOAD_DIR) $(BUILD_DIR)
+$(EXTRACT): $(DOWNLOAD) | $(DOWNLOAD_DIR) $(BUILD_DIR)
 	$(if $(filter %zip,$(pkg_ext)), \
 		unzip -q $(DOWNLOAD_DIR)/$(pkg_archive_name) -d $(BUILD_DIR), \
 		tar -xf $(DOWNLOAD_DIR)/$(pkg_archive_name) -C $(BUILD_DIR))
@@ -92,7 +92,7 @@ $(EXTRACT): | $(DOWNLOAD_DIR) $(BUILD_DIR)
 PATCH  := $(BUILD_DIR)/.patched
 patch : $(PATCH)
 PKG_PATCHES ?=
-$(PATCH): $(PKG_PATCHES) | $(BUILD_DIR)
+$(PATCH): $(EXTRACT) $(PKG_PATCHES) | $(BUILD_DIR)
 	if [ -d tree ]; then \
 		cd tree; \
 		cp -l -u -f --parents -t $(PKG_SOURCE_DIR) `find . -not -path '*/.*' -and -type f`; \
@@ -105,15 +105,15 @@ $(PATCH): $(PKG_PATCHES) | $(BUILD_DIR)
 
 CONFIGURE  := $(BUILD_DIR)/.configured
 configure : $(CONFIGURE)
-$(CONFIGURE): | $(BUILD_DIR)
+$(CONFIGURE): $(PATCH) | $(BUILD_DIR)
 
 BUILD  := $(BUILD_DIR)/.builded
 build : $(BUILD)
-$(BUILD): | $(BUILD_DIR)
+$(BUILD): $(CONFIGURE) | $(BUILD_DIR)
 
 INSTALL  := $(BUILD_DIR)/.installed
 install : $(INSTALL)
-$(INSTALL): | $(BUILD_DIR) $(PKG_INSTALL_DIR)
+$(INSTALL): $(BUILD) | $(BUILD_DIR) $(PKG_INSTALL_DIR)
 
 # Definitions used by user Makefile
 

--- a/src/arch/usermode86/host/host_net.c
+++ b/src/arch/usermode86/host/host_net.c
@@ -12,7 +12,6 @@
 #include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
-#include <stropts.h>
 
 #include <net/if.h>
 #include <linux/if_tun.h>

